File: 1974-minimum-time-to-type-word-using-special-typewriter/1974-minimum-time-to-type-word-using-special-typewriter.java
class Solution {
    public int minTimeToType(String word) {
        int result=word.length();
        char prev='a';
        
        for(int i=0;i<word.length(); i++)
        {
            char curr=word.charAt(i);
            int step=Math.abs(prev-curr);
            result+=Math.min(step,26-step);
            prev=curr;
        }
        
        return result;
    }
}

File: 1974-minimum-time-to-type-word-using-special-typewriter/1974-minimum-time-to-type-word-using-special-typewriter.cpp
class Solution {
public:
    int minTimeToType(string word) {
        char current='a';
        int result=0;
        
        for(int i=0; i<word.size();i++)
        {
            int dis=abs(word[i]-current);
            int sec=min(dis,26-dis);
            result+=(sec+1);
            current=word[i];
        }
        return result;
    }
};